modules/react/text-input/spec/InputGroup.spec.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import {renderToString} from 'react-dom/server';
+
+import {searchIcon, xSmallIcon} from '@workday/canvas-system-icons-web';
+import {InputGroup} from '../lib/InputGroup';
+import {TertiaryButton} from '@workday/canvas-kit-react/button';
+import {SystemIcon} from '@workday/canvas-kit-react/icon';
+
+describe('InputGroup', () => {
+  verifyComponent(InputGroup, {});
+
+  it('should render on a server without crashing', () => {
+    const ssrRender = () =>
+      renderToString(
+        <InputGroup>
+          <InputGroup.Start pointerEvents="none">
+            <SystemIcon icon={searchIcon} size="small" />
+          </InputGroup.Start>
+          <InputGroup.Input />
+          <InputGroup.End>
+            <TertiaryButton role="presentation" icon={xSmallIcon} size="small" tabIndex={-1} />
+          </InputGroup.End>
```

Comment:
Seems like adding a `tabIndex={-1}` is going to be an important caveat for teams.

Reply:
I was thinking about introducing a `InputGroup.ClearButton` that has this built in, including clearing the input when clicked.

We don't have specification for it yet. The Preview combobox has this button as focusable. I've also seen this be a non-focusable button because there's already keyboard functionality to clear an input. In my opinion, it should not be focusable via the tab keys. Our accessibility SMEs tend to agree on less focus stops of non-important elements. It makes the interface less predictable.

Reply:
I think this is fine for now. Just seems like an area we could improve in the future.